off-chain/test/Examples/Addition/ContractSeath.purs

```diff
@@ -1,101 +1,93 @@
 module Seath.Test.Examples.Addition.ContractSeath (mainTest) where
 
 import Contract.Log (logInfo')
-import Contract.Monad (Aff, Contract, throwContractError)
-import Contract.Prelude (when, (/=), (>>=))
+import Contract.Monad (Contract, throwContractError)
+import Contract.Prelude (unless, unwrap, when, (/=), (>>=))
 import Contract.Scripts (ValidatorHash)
-import Contract.Test.Plutip (PlutipConfig, runPlutipContract)
 import Contract.Transaction (awaitTxConfirmed)
 import Contract.Wallet (withKeyWallet)
 import Control.Monad (bind)
-import Data.Array (last, length, replicate, unzip)
-import Data.BigInt (BigInt)
-import Data.BigInt as BigInt
+import Data.Array (last, unzip)
+import Data.Array.NonEmpty as NE
 import Data.Either (Either, note)
 import Data.Functor (map)
 import Data.List (head)
 import Data.Map (size, values)
 import Data.Maybe (Maybe(Just, Nothing), maybe)
 import Data.Monoid ((<>))
 import Data.Show (show)
-import Data.Tuple.Nested (type (/\), (/\))
+import Data.Tuple.Nested ((/\))
 import Data.Unit (Unit)
 import Prelude (discard, pure, ($))
 import Seath.HandleActions (buildChain)
-import Seath.Test.Examples.Addition.Actions
-  ( fixedValidatorHash
-  , handleAction
-  , queryBlockchainState
-  )
+import Seath.Test.Examples.Addition.Actions (fixedValidatorHash, handleAction, queryBlockchainState)
 import Seath.Test.Examples.Addition.Contract (initialSeathContract)
-import Seath.Test.Examples.Addition.SeathSetup
-  ( BlockhainState(BlockhainState)
-  , Leader(Leader)
-  , Participant(Participant)
-  , getBlockhainState
-  , getPublicKeyHash
-  , logBlockchainState
-  )
+import Seath.Test.Examples.Addition.SeathSetup (BlockhainState(BlockhainState), Leader(Leader), Participant(Participant), getBlockhainState, getPublicKeyHash, logBlockchainState)
 import Seath.Test.Examples.Addition.SeathSetup as SeathSetup
-import Seath.Test.Examples.Addition.Types (AdditionDatum(AdditionDatum))
+import Seath.Test.Examples.Addition.Types (AdditionDatum)
 import Seath.Test.Examples.Utils (getTypedDatum)
+import Seath.Test.TestSetup (RunnerConfig(RunnerConfig), runnerConfInfo)
 import Seath.Types (SeathConfig(SeathConfig))
 
-mainTest :: PlutipConfig -> Aff Unit
-mainTest config = runPlutipContract config distribution $
-  \((admin /\ leader') /\ participants') -> do
-    -- contract initialization by some admin
-    _ <- withKeyWallet admin initialSeathContract
+mainTest :: RunnerConfig AdditionDatum -> Contract Unit
+mainTest config = do
+  -- todo: check that parties participants have enough funds by config.minAdaRequired
+  logInfo' $ "Running with " <> runnerConfInfo config
+
+  let
+    leaderKeyWallet = (unwrap config).seathLeader
+    leader = Leader leaderKeyWallet
 
+  unless ((unwrap config).alreadyInitialized) $ do
+    -- contract initialization by some admin
+    _ <- withKeyWallet leaderKeyWallet initialSeathContract
```

Comment:
So, we lock admin funds at every running on testnet?

Reply:
Oh, this is an error with wallet I've used leader's wallet insted of admin's.
The intention is to control initiation by admin, so for the testnet next run won't break check [here](https://github.com/mlabs-haskell/seath/blob/5ce08dca26e323ddab324ceb0974e2c5604104c5/off-chain/test/Examples/Addition/Actions.purs#L68).

For prperpod `alreadyInitialized` set to `true`, so it shouldn't run now, for plutip - `false`, coz we run from scratch every time.
Thinking now, maybe better pass it with args?

Reply:
What about querying the `scriptUtxos` and if there isn't any done by the `admin` then create one?

Reply:
yes, we definitely need to query the UTxO in the script as we need to know the current state, this way we can calculate the final expected result instead of having it in the config (test if failing at the end because the expected value is hardcoded).

Reply:
I've added check for state change which uses starting and ending states to verify expected changes.
Let me know if it looks good to you.

Reply:
> What about querying the scriptUtxos and if there isn't any then create one?
Also added.